app/services/redirection_creation.rb

```diff
@@ -0,0 +1,25 @@
+class RedirectionCreation
+  def initialize(referrer, slug)
+    @referrer = referrer
+    @slug = slug
+  end
+
+  def perform
+    redirection = Redirection.new(slug: slug)
+
+    if referrer.present?
+      redirection.url = referrer_hostname
+      Ring.new(redirection).link
+      redirection
+    end
+  end
+
+  private
+
+  attr_reader :referrer, :slug
+
+  def referrer_hostname
+    referrer_uri = URI.parse(referrer)
```

Comment:
I only knew about `URI(referrer)`. `.parse` is probably slightly more clear anyway.